jsnlog.ts

```diff
@@ -1079,6 +1101,13 @@ module JL
                         }
                     }
 
+                    if (typeof JL.traceContextProvider === 'function') {
```

Comment:
Please change this so the existing trace context (if any) is passed into the traceContextProvider method. This way, that method can modify any existing trace context, rather than having to completely replace an existing trace context with a new one. That way the method could for example only update the trace id inside the trace context, while leaving the other fields the same.

So the signature of that method would be something like
(JSNLogTraceContext)=>JSNLogTraceContext
instead of
()=>JSNLogTraceContext